src/binary-search-tree.js

```diff
@@ -5,6 +5,68 @@ class BinarySearchTree {
     this.left = null;
     this.right = null;
   }
+  insert(data) {
+    const search = (node) => {
+      if (data < node.value && node.left) search(node.left);
+      else if (data < node.value) node.left = new BinarySearchTree(data);
+      else if (data > node.value && node.right) search(node.right);
+      else if (data > node.value) node.right = new BinarySearchTree(data);
+    };
+    search(this);
+  }
+  contains(data) {
+    let flag = false;
+    const search = (node) => {
+      if (data === node.value) flag = true;
```

Comment:
Same here, you'll want to return after setting the flag so that your recursive function doesn't continue executing after it's already found the target data.